test/testdata/parser/error_recovery/unterminated_array.rb.parse-tree.exp

```diff
@@ -0,0 +1,133 @@
+Begin {
```

Comment:
I noticed that in this PR as well as #5272 you used `.parse-tree.exp`. I've been using `desugar-tree` for all my parser recovery tests because I find them much easier to tell at a glance whether they're correct. For example, it's easy for me to tell whether something is in one method body or has been pushed outside it.

I'm curious: do you have a strong preference for `parse-tree` tests? I'd be curious to hear more, or if it's not a strong preference I'd love to see desugar-tree tests be the default here.

Reply:
All the tests in the `error_recovery` dir use `parse-tree`, so I was just following the pattern here and in #5272.